app/controllers/bids_controller.rb

```diff
@@ -31,13 +30,14 @@ def new
   end
 
   def confirm
-    @auction = Presenter::Auction.new(Auction.find(params[:auction_id]))
-    @bid = Presenter::Bid.new(PlaceBid.new(params, current_user).dry_run)
+    @auction = Policy::Auction.factory(Auction.find(params[:auction_id]), current_user)
+    @bid = Policy::Bid.new(PlaceBid.new(@auction, params).dry_run, @auction)
```

Comment:
The PlaceBid#dry_run in the middle of a new bid seems funky. It reads as really complex ... making a new bid policy out of a fake place bid activity??? Probably some code is is the wrong place, and rearranging code is what design is all about!